nightfall-deployer/src/contract-setup.mjs

```diff
@@ -14,26 +14,19 @@ async function setupContracts() {
   const stateInstance = await waitForContract('State');
   logger.debug(`address of State contract is ${stateInstance.options.address}`);
 
-  const tx = {
-    from: process.env.FROM_ADDRESS,
-    // value: fee,
-    gas: config.WEB3_OPTIONS.gas,
-    gasPrice: config.WEB3_OPTIONS.gasPrice,
-  };
-
-  const proposers = await waitForContract('Proposers');
-  const shield = await waitForContract('Shield');
-  const challenges = await waitForContract('Challenges');
+  const proposersInstance = await waitForContract('Proposers');
+  const shieldInstance = await waitForContract('Shield');
+  const challengesInstance = await waitForContract('Challenges');
 
   const simpleMultiSigAddress = (await waitForContract('SimpleMultiSig')).options.address;
 
   const contracts = {
-    proposers,
-    shield,
-    challenges,
+    proposersInstance,
+    shieldInstance,
+    challengesInstance,
   };
 
-  for await (const contractName of ['proposers', 'shield', 'challenges']) {
+  for await (const contractName of Object.keys(contracts)) {
```

Comment:
I don't know if here we could parallelized with `Promise.all` instead of `for await` like this and also avoid then `Object.keys`
```
  await Promise.all(
    contracts.map(async contract => {
      const setStateContract = contract.methods.setStateContract(stateInstance.options.address);
      const transferOwnership = contract.methods.transferOwnership(simpleMultiSigAddress);
      if (!config.ETH_PRIVATE_KEY) {
        setStateContract.send();
        transferOwnership.send();
      } else {
        await Web3.submitRawTransaction(setStateContract.encodeABI(), contract.options.address);
        await Web3.submitRawTransaction(transferOwnership.encodeABI(), contract.options.address);
      }
    }),
  );
```

Reply:
I tried that way previously, and it works. But after some research I found the modern `for await` and I believe it makes the code easier to read than `Promise.all`. It's a styling matter, I can make the change back to `Promise.all` if you prefer

Reply:
What I found is this

"Use for await of if the order in which promises are resolved is important to you. Use Promise. all() if the order isn't important and you need all calls to succeed."

Reply:
Oh I thought `for await` would also parallelize it just the same. And actually after refactoring to your code it looks more concise and understandable. I'll push a new commit with this change, thanks man